spec/integration/rails/transaction_hooks_spec.rb

```diff
@@ -230,13 +238,31 @@ def json
         it "fires all before and after_commit hooks" do
           post :create, params: payload
           expect(Callbacks.entities).to eq([
+            :before_validation,
             :before_create,
             :stacked_before_create,
             :employee_after_create,
             :employee_after_create_eval_test,
           ])
         end
       end
+
+      context "when an error is raised before_validation" do
+        before do
+          $raise_on_before_validation = {employee: true}
+        end
+
+        it "does not run before_commit callbacks" do
+          expect_any_instance_of(Graphiti::Util::ValidationResponse)
```

Comment:
You did a great job with the general setup, but I think this is the issue that would hold up the PR. We wanted something to fire ***before*** validation. As you can see and line 126 of `resource_proxy.rb`, this is happening ***after*** validation. IOW, we want `to_not receive` - and probably a Rails integration spec that addresses the [original issue](https://github.com/graphiti-api/graphiti/issues/93).

I don't think you're too far off though! Hope we get this in 👏 !